contrib/mesos/pkg/scheduler/podtask/port_mapping.go

```diff
@@ -99,16 +99,16 @@ func wildcardHostPortMapping(t *T, offer *mesos.Offer) ([]HostPortMapping, error
 	remaining := len(wildports)
 	foreachRange(offer, "ports", func(bp, ep uint64) {
 		log.V(3).Infof("Searching for wildcard port in range {%d:%d}", bp, ep)
-		for _, entry := range wildports {
-			if entry.OfferPort != 0 {
+		for i := range wildports {
+			if wildports[i].OfferPort != 0 {
```

Comment:
I see the loop has been changed to avoid copying `wildports[i]` into `entry`. I'm not sure how/why this fixes the problem that you've identified because in the code (a) `entry` is copied into `mapping` via `append`, and; (b) we don't reference `wildports` after the loop completes.